satellite/gracefulexit/endpoint.go

```diff
@@ -535,38 +426,21 @@ func (endpoint *Endpoint) processIncomplete(ctx context.Context, stream processS
 		if err != nil {
 			return Error.Wrap(err)
 		}
-	}
-	remote := pointer.GetRemote()
-	pieces := remote.GetRemotePieces()
-
-	var nodePiece *pb.RemotePiece
-	excludedNodeIDs := make([]storj.NodeID, len(pieces))
-	for i, piece := range pieces {
-		if piece.NodeId == nodeID && piece.PieceNum == incomplete.PieceNum {
-			nodePiece = piece
-		}
-		excludedNodeIDs[i] = piece.NodeId
-	}
-
-	if nodePiece == nil {
-		endpoint.log.Debug("piece no longer held by node", zap.Stringer("Node ID", nodeID), zap.ByteString("path", incomplete.Path), zap.Int32("piece num", incomplete.PieceNum))
-
-		err = endpoint.db.DeleteTransferQueueItem(ctx, nodeID, incomplete.Path, incomplete.PieceNum)
-		if err != nil {
-			return Error.Wrap(err)
-		}
 
 		return nil
 	}
 
-	redundancy, err := eestream.NewRedundancyStrategyFromProto(pointer.GetRemote().GetRedundancy())
+	nodePiece, err := endpoint.getNodePiece(ctx, pointer, incomplete)
```

Comment:
Do you think it would be worth combining `getNodePiece` and `getValidPointer` in the future? We don't have to do it here, but it seems like it might make sense to combine them to me.